src/fios.cpp

```diff
@@ -300,13 +300,29 @@ bool FiosFileScanner::AddFile(const char *filename, size_t basepath_length, cons
 
 	FiosItem *fios = file_list.Append();
 #ifdef _WIN32
-	struct _stat sb;
-	if (_tstat(OTTD2FS(filename), &sb) == 0) {
+	// Retrieve the file modified date using GetFileTime rather than stat to work around an obscure MSVC bug that affects Windows XP
+	HANDLE fh = CreateFileW(OTTD2FS(filename), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE, nullptr, OPEN_EXISTING, 0, nullptr);
```

Comment:
Unless we're straight up abandoning the ability to do ANSI (non-UNICODE) builds this should be `CreateFile` with neither A nor W suffix.

Reply:
Ah, good point - for some reason I thought we'd already done that. (Or maybe it's just that all my C++ stuff at work is Unicode-only and has been for years!)